### PR TITLE
[18.09] remove html tags and newlines from the help text

### PR DESCRIPTION
**Pull request by martenson**
this should improve the toolsearch for certain keywords
and also make the search index smaller

### CODE REVIEW

File: lib/galaxy/tools/search/__init__.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 from galaxy.web.framework.helpers import to_unicode
 
 log = logging.getLogger(__name__)
+HTML_CLEANER = re.compile("<[^<]+?>")
```
Comment (martenson) [inline lib/galaxy/tools/search/__init__.py]: poor man's beautiful soup/lxml, please offer better alternatives if you have

Comment (nsoranzo) [reply lib/galaxy/tools/search/__init__.py]: https://bleach.readthedocs.io/en/latest/ could be an option, since we already use it elsewhere.

Comment (martenson) [reply lib/galaxy/tools/search/__init__.py]: Do you have any experience with its speed?

Comment (martenson) [reply lib/galaxy/tools/search/__init__.py]: so it does not really behave as I want it to:

```python
>>> import bleach
>>> bleach.clean('test<strong>sdsd</strong>', strip=True)
u'test<strong>sdsd</strong>'
>>> bleach.clean('test<span></span>', strip=True)
u'test'
```

Comment (martenson) [reply lib/galaxy/tools/search/__init__.py]: it seems I need to define an empty list of allowed tags, then it works well, commit inc.

**Review by nsoranzo — APPROVED** [review]

**Comment by martenson** [comment]
just for fun: previously you could search for html tag contents, eg `strong` and you would get the tools that use the most bold font :D

screenshot 2018-09-18 09 48 12

**Pull request merged**